### PR TITLE
So2 bug fix

### PR DESCRIPTION
**Pull request by cjpurackal**
#### Changes

Fixes some bugs associated with batched/unbatched complex data

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/geometry/liegroup/so2.py [hunk 1]
```diff
@@ -187,7 +185,11 @@ def from_matrix(cls, matrix: Tensor) -> 'So2':
             or (len_matrix_shape > 3 or len_matrix_shape < 2)
         ):
             raise ValueError(f"Invalid input size, we expect [B, 2, 2] or [2, 2]. Got: {matrix_shape}")
-        return cls(complex(matrix[..., 0, 0], matrix[..., 1, 0]))
+        if len_matrix_shape == 2:
+            z = complex(matrix[0, 0], matrix[1, 0])
```
Comment (edgarriba) [inline kornia/geometry/liegroup/so2.py:189]: Even with unbatched tensors the ellipsis will work. So no need the branching

File: kornia/geometry/liegroup/so2.py [hunk 2]
```diff
@@ -200,24 +202,23 @@ def identity(cls, batch_size: Optional[int] = None, device=None, dtype=None) ->
...
+            real_data = real_data.repeat(batch_size)
+            imag_data = imag_data.repeat(batch_size)
         return cls(complex(real_data, imag_data))
 
     def inverse(self) -> 'So2':
         """Returns the inverse transformation.
 
         Example:
-            >>> s = So2.identity(batch_size=1)
+            >>> s = So2.identity(1)
```
Comment (edgarriba) [inline kornia/geometry/liegroup/so2.py:219]:
```suggestion
            >>> s = So2.identity()
```

File: test/geometry/liegroup/test_so2.py [hunk 3]
```diff
@@ -17,27 +17,27 @@ def test_smoke(self, device, dtype):
         assert isinstance(s, So2)
         self.assert_close(s.z.data, z.data)
 
-    @pytest.mark.parametrize("input_shape", [(1, 1), (2, 1), (5, 1), (1,), (2,), (5,)])
+    @pytest.mark.parametrize("input_shape", [(1,), (2,), (5,), ()])
```
Comment (edgarriba) [inline test/geometry/liegroup/test_so2.py:20]: Add a None batched version too tensor (1) != tensor([1])

File: kornia/geometry/liegroup/so2.py [hunk 4]
```diff
@@ -148,21 +146,21 @@ def hat(theta: Tensor) -> Tensor:
         theta = theta.reshape(-1, 1)
         row0 = pad(theta, (1, 0))
         row1 = pad(theta, (0, 1))
-        return stack((row0, row1), -1)
+        return stack((row0, row1), -1).squeeze()
```
Comment (edgarriba) [inline kornia/geometry/liegroup/so2.py:149]: Don’t think squeeze is necessary. Otherwise there’s a bad assumption somewhere

**Review by edgarriba — CHANGES_REQUESTED** [review]

File: test/geometry/liegroup/test_so2.py (hunk 3 above)
Comment (edgarriba) [reply test/geometry/liegroup/test_so2.py:20]: Oh, you have () — nevermind

File: kornia/geometry/liegroup/so2.py (hunk 2 above)
Comment (cjpurackal) [reply kornia/geometry/liegroup/so2.py]: for some reason, __repr__ doesn't want to print out the full string 
`Parameter containing:
 tensor(1.+0.j, requires_grad=True)` when unbatched instead only prints `(1+0j)`.
 Prints the full string when i add in .z

File: kornia/geometry/liegroup/so2.py [hunk 5]
```diff
@@ -143,10 +143,10 @@ def hat(theta: Tensor) -> Tensor:
             or (len_theta_shape > 2)
         ):
             raise ValueError(f"Invalid input size, we expect [B, 1] or [B]. Got: {theta_shape}")
-        theta = theta.reshape(-1, 1)
+        theta = theta.reshape(-1, 1) if any(theta.shape) else theta.reshape(-1)
```
Comment (edgarriba) [inline kornia/geometry/liegroup/so2.py:146]: not sure if its more clean this
```python
z = zeros_like(theta)
row0 = stack((z, theta), -1)
row1 = stack((theta, z), -1)
return stack((row0, row1), -1)
```

**Review by edgarriba — APPROVED** [review]

**Pull request merged**